### PR TITLE
(OraklNode) Update context usage from raft

### PR DESCRIPTION
**Pull request by nick-bisonai**
# Description

Raft package is to be used from aggregator package and submitter package.
Through this update, it will more utilize the context given from utilizing packages, and stop if `ctx.cancel()` is called from utilizing packages.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.

## Deployment

- [ ] Should publish npm package
- [ ] Should publish Docker image

### CODE REVIEW

File: node/pkg/raft/raft.go [hunk 1]
```diff
@@ -60,28 +62,34 @@ func (r *Raft) subscribe(ctx context.Context) {
 		log.Error().Err(err).Msg("failed to subscribe to topic")
 	}
 	for {
+		if ctx.Err() == context.Canceled {
```
Comment (martinkersner) [inline node/pkg/raft/raft.go:65]: It is maybe too late to say this, but isn't there a solution without an infinite loop? 👀

**Review by martinkersner — APPROVED** [review]
LGTM!

File: node/pkg/raft/raft.go (hunk 1 above)
Comment (nick-bisonai) [reply node/pkg/raft/raft.go:65]: 🤔 can you give me more detail? it should continously get message from other raft peers so I put it inside infinite poop

Comment (nick-bisonai) [reply node/pkg/raft/raft.go:65]: I thought it was blocking call sonce the official example code also implemented it through infinite for loop

'''
func printMessagesFrom(ctx context.Context, sub *pubsub.Subscription) {
	for {
		m, err := sub.Next(ctx)
		if err != nil {
			panic(err)
		}
		fmt.Println(m.ReceivedFrom, ": ", string(m.Message.Data))
	}
}
'''
https://github.com/libp2p/go-libp2p/blob/master/examples/pubsub/basic-chat-with-rendezvous/main.go

Comment (martinkersner) [reply node/pkg/raft/raft.go:65]: Got it! So the `subscribe` function is called through go routine?

Comment (nick-bisonai) [reply node/pkg/raft/raft.go:65]: yes, you can refer to line 40

Comment (martinkersner) [reply node/pkg/raft/raft.go:65]: Thank you! I guess it is good to keep, but I am surprised they haven't implement it with channels instead.

Comment (nick-bisonai) [reply node/pkg/raft/raft.go:65]: 🤔 i'll double check official docs tmr

**Pull request merged**